rs/execution_environment/src/query_handler.rs

```diff
@@ -248,6 +248,8 @@ impl InternalHttpQueryHandler {
         // Letting the canister grow arbitrarily when executing the
         // query is fine as we do not persist state modifications.
         let subnet_available_memory = subnet_memory_capacity(&self.config);
+        // XXX Should we allow more than the canister's quota of downstream calls?
```

Comment:
Is "XXX" typo?

Reply:
No, it's a marker intended to attract attention. Which I suppose it did. (o:

This is actually a question for you guys: Should we allow a composite query to make 1M downstream calls? Set a lower limit? Should that limit be the same as the guaranteed quota?

I have no clue what the limitations of composite queries are / should be.

Reply:
My current feeling is that a bound (much) lower than 1M would be good to have. If we'd set it to be the same as the guaranteed quota (i.e., 50) I'm pretty sure we'd not constrain any use cases.

Reply:
That is my gut feeling too, but I wanted to make sure I wasn't missing anything. Hence the `XXX`.

Reply:
I do not have enough context on this to answer, @dsarlis WDYT?

Reply:
Hmm, that's an interesting one. I think we might want to be a bit stricter with composite queries limits but on the other hand we should also not have some behaviour be different when running in replicated vs non-replicated mode. I think we even mention some of the composite query limits in the spec. @mraszyk would you have an opinion here (both on what a reasonable limit would be and/or whether it's worth covering in the spec)?

Reply:
Basically, think that you have 5B instructions to execute your whole query graph in a composite query (same as a single query). So, allowing 1M downstream calls in the composite query context likely does not make sense anyway (I imagine you'd hit the instruction limit way sooner than being able to perform these 1M downstream calls).

Reply:
I don't really understand how we enforce the instruction limit (are all queries in the call tree executed sequentially?).

But if 5B instructions is the limit we aim for and enforce, then as you say, it doesn't matter whether we allow 1M callbacks or more, we're not going to get anywhere near that.

Reply:
> I don't really understand how we enforce the instruction limit (are all queries in the call tree executed sequentially?).

Well, in a sense, yeah. We use a depth-first search algorithm to evaluate and go through the entire query graph (yes, essentially locking up the thread for the duration). Did I mention I do _not_ like composite queries? :P 

> But if 5B instructions is the limit we aim for and enforce, then as you say, it doesn't matter whether we allow 1M callbacks or more, we're not going to get anywhere near that.

I would also be fine with leaving it up to the instruction limit to prevent misuse. It's there for a reason.

Reply:
I'll just leave it at the current 1M then, unless someone suggests a lower number and a reason for it.